doc/design/new-user-questionnaire.md

```diff
@@ -0,0 +1,56 @@
+# Premise
+Currently, anyone with a GitHub account can signup for DANDI and make full use of its functionality. The goal of this design document is to introduce an
+approval process to help prevent misuse of the archive.
+
+## 1. Implementation Details
+### Changes to User model
+Three new columns, `status`, `questionnaire_form`, and `rejection_reason` will be introduced to the `User` model in dandi-api.
+- `status` is an enum that can be one of four values: `INCOMPLETE`, `PENDING`, `APPROVED`, or `REJECTED`.
+- `questionnaire_form` is a JSON field. The keys are questions, and the values are the users' answers.
+- `rejection_reason` is an optional string that contains a message written by one of the admins describing why a user was denied approval (if applicable).
+
+### Changes to API
+- /users/me needs to include the user's current status
+- A [Django group](https://docs.djangoproject.com/en/3.2/topics/auth/default/#groups), `approved_users`, will be created. Only users in this group will have access to protected endpoints.
+- Only approved users can create and edit dandisets.
+
+### Changes to frontend
+- Configuration so the frontend knows where to find SSR pages
+- User avatar circley thing needs to show user's current status
+
+### New views
+- SSR rendered pages for questionnaire form, pending approval page, rejected page, and admin approval page.
```

Comment:
Not using SSR seems quite complex, when the user creation and verification procedures *already* "bind web ui and dandi-api".

@mvandenburgh can provide further detail on why SSR is needed here, but I don't think separation of concerns is a valid argument in this case. The users are being created and processed serverside, after all.

Reply:
Everything pretty much is being created and processed server side, while ui provides parameters etc.

Having more of such interactions going not through API and main web ui IMHO
- hinders UX (some custom web forms outside of the main web ui look/feel)
- might later manifest itself in other locations (testing etc) requiring more selenium style navigation instead of just a few API calls

Reply:
I didn't really answer the original question: indeed, we do use SSR pages for the login flow. In particular, the "authorize DANDI?" page that we suppressed in deployment was an SSR page, which is appropriate because everything in the login flow happens very much serverside, without any participation from the client.

As far as UX, using something like Materialize CSS should easily match the style of the app. And as for testing, we would indeed use end-to-end testing for this, but since we're testing an extra component of the login flow, that would be an appropriate mode of testing.

In the end, doing this as an SSR page will simplify the development of this feature and avoid adding extra, unnecessary complexity to the API. It also matches the engineers' intuition about how this will work, so I think it's a win.

Reply:
ok, SSR it is